Source/hu_stuff.c

```diff
@@ -805,25 +805,67 @@ static int HU_top(int i, int idx1, int top1)
 
 static void HU_widget_build_monsec(void)
 {
+  int i, playerscount;
   char *s;
+  char kills_str[60];
   int offset = 0;
 
-  int killcolor = (plr->killcount - extrakills >= totalkills ? '0'+CR_BLUE : '0'+CR_GOLD);
-  int kill_percent_color = (plr->killcount >= totalkills ? '0'+CR_BLUE : '0'+CR_GOLD);
-  int kill_percent = (totalkills == 0 ? 100 : plr->killcount * 100 / totalkills);
-  int itemcolor = (plr->itemcount >= totalitems ? '0'+CR_BLUE : '0'+CR_GOLD);
-  int secretcolor = (plr->secretcount >= totalsecret ? '0'+CR_BLUE : '0'+CR_GOLD);
+  int kills = 0, kills_color, kills_percent, kills_percent_color;
+  int items = 0, items_color;
+  int secrets = 0, secrets_color;
+
+  for (i = 0, playerscount = 0; i < MAXPLAYERS; ++i)
+  {
+    int color = (i == displayplayer) ? '0'+CR_GRAY : '0'+CR_GOLD;
+    if (playeringame[i])
+    {
+      if (playerscount == 0)
+      {
+        offset = sprintf(kills_str,
+          "\x1b%c%d", color, players[i].killcount);
+      }
+      else
+      {
+        offset += sprintf(kills_str + offset,
```

Comment:
Isn't `offset` definitely still `0` here?

Reply:
No, in this case there are more than 1 players, so "+killcount" is added to the string.

Reply:
Ah, okay, I have overseen this is within a loop.